scipy/conftest.py

```diff
@@ -233,23 +262,25 @@ def skip_or_xfail_xp_backends(xp, backends, kwargs, skip_or_xfail='skip'):
         but not all, non-CPU/non-NumPy backends.
     """
     skip_or_xfail = getattr(pytest, skip_or_xfail)
-
     np_only = kwargs.get("np_only", False)
     cpu_only = kwargs.get("cpu_only", False)
     exceptions = kwargs.get("exceptions", [])
     
+    if reasons := kwargs.get("reasons"):
+        raise ValueError(f"provide a single `reason=` kwarg; got {reasons=} instead")
+
     # input validation
     if np_only and cpu_only:
-        raise ValueError("at most one of `np_only` and `cpu_only` should be provided")
+        # np_only is a stricter subset of cpu_only
+        cpu_only = False
```

Comment:
is it necessary to set `cpu_only = False`?

Reply:
Hard to tell TBH, esp without the full CI giving a definitive yay/nay. I certainly had issues while iterating where both variables were `True`. Might be that it was a transient thing. 

I'm not set to switch over to pixi right now, so if you do have it set up --- could you try removing the whole stanza and rerun the full battery of tests? And push if it's indeed OK.

Reply:
I'm happy to just leave it instead of re-running all of the tests, but pretty sure that the value of `cpu_only` becomes redundant in the following code when `np_only` is `True`.

Reply:
Let's leave as is for now then?